### ISSUE TITLE
Question about training

### ISSUE THREAD

**Issue by yiranxie233** [VuBacktracking/mamba-text-classification#1]
Hello, thank you giving the project about how to use mamba model for text classification! 
When I use your code for training, I find a problem: the more training step forward, the more GPU memory needed. Is that the character of the mamba model store the hiddden state when training?

**Comment by VuBacktracking** [comment]
Thanks for your question.
I'd be curious to know if you're training the model on a GPU? If so, what's the size of your GPU memory?

**Comment by yiranxie233** [comment]
Yes, I trained on RTX3060, and the size of my GPU is 6GB, I first try train batch_size=2, then out of GPU memory, then I put train batch_size=1, then completed the train.

**Comment by VuBacktracking** [comment]
Yeah, that would be great. Reducing the batch size helped you complete training on your model. This is a great strategy for smaller GPUs. Additionally, gradient checkpointing can further reduce memory usage.
